pkg/crc/machine/start.go

```diff
@@ -1070,6 +1067,19 @@ func startMicroshift(ctx context.Context, sshRunner *crcssh.Runner, ocConfig oc.
 	return cluster.WaitForAPIServer(ctx, ocConfig)
 }
 
+func ensurePullSecretPresentInVM(sshRunner *crcssh.Runner, pullSec cluster.PullSecretLoader) error {
+	if pullSecret, _, err := sshRunner.RunPrivileged("Checking if pull secret already present in the VM", "cat", "-f", "/etc/crio/openshift-pull-secret"); err == nil {
+		if err := validation.ImagePullSecret(pullSecret); err == nil {
+			return nil
+		}
+	}
+	content, err := pullSec.Value()
```

Comment:
Maybe add a comment before this since it's reversed compared to usual go code which exit early on errors, not on success:
```
// pull secret is not present in VM or is invalid
```